src/target/llvm/intrin_rule_hexagon.cc

```diff
@@ -19,44 +19,54 @@
 
 #ifdef TVM_LLVM_VERSION
 
+#include <tvm/tir/op_attr_types.h>
+
 #include "intrin_rule_llvm.h"
 
 namespace tvm {
 namespace codegen {
 namespace llvm {
+using tir::FLowerIntrinsic;
 
-TVM_REGISTER_GLOBAL("tvm.intrin.rule.hexagon.exp")
-    .set_body(DispatchLLVMPureIntrin<::llvm::Intrinsic::exp, 1>);
+TVM_REGISTER_OP("tir.exp").set_attr<FLowerIntrinsic>(
+    "hexagon.FLowerIntrinsic", PackedFunc(DispatchLLVMPureIntrin<::llvm::Intrinsic::exp, 1>));
```

Comment:
would be nice if we can modify the signature of DispatchLLVMPureIntrin to be PrimExpr->PrimExpr, so we do not have to do the convertion to PackedFunc

Reply:
Thanks! Fixed in the new commit.